src/lupapiste_commons/tos_metadata_schema.cljc

```diff
@@ -155,11 +155,13 @@
          (filter (fn [[k _]] (known-keys k)))
          (into {}))))
 
-(defn remove-conditional-keys [{:keys [sailytysaika julkisuusluokka] :as metadata}]
+(defn remove-conditional-keys [{:keys [sailytysaika julkisuusluokka permit-expired demolished] :as metadata}]
   (cond-> metadata
           (not= (:arkistointi sailytysaika) :määräajan)    (dissoc-in [:sailytysaika :pituus])
           (not= (:arkistointi sailytysaika) :määräajan)    (dissoc-in [:sailytysaika :retention-period-end])
           (not= (:arkistointi sailytysaika) :toistaiseksi) (dissoc-in [:sailytysaika :laskentaperuste])
+          (false? permit-expired)                          (dissoc-in [:permit-expired-date])
+          (false? demolished)                              (dissoc-in [:demolished-date])
```

Comment:
Tää on sikäli ehkä väärä paikka, että nämä avaimet eivät ole tässä skeemassa määriteltyjä, mutta eipä kait sillä loppujen lopuksi väliä, kun ei näiden avainten puuttuminen haittaa muualla.

`dissoc-in` voisi olla `dissoc` kun tarvitsee vain päätasolta dissocata.